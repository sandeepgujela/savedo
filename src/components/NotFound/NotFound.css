.NotFound {
  text-align: center;
  font-size: 24px;
  color: #5ab
}
