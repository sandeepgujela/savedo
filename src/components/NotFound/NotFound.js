import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './NotFound.css';

class NotFound extends Component {
  render() {
    const { prop = 'Not Found' } = this.props;
    return (
      <div className="NotFound">
        <span>{prop}</span>
      </div>
    );
  }
}

NotFound.propTypes = {
  prop: PropTypes.string.isRequired,
};

export default NotFound;
