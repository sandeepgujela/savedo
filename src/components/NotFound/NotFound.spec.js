import React from 'react';
import { shallow } from 'enzyme';
import NotFound from './NotFound';

describe('<NotFound />', () => {
  const build = passedProps => {
    const defaultProps = { prop: 'NotFound' };
    return shallow(<NotFound {...defaultProps} {...passedProps} />);
  };

  it('should render', () => {
    const component = build();
    expect(component).toMatchSnapshot();
  });
});
