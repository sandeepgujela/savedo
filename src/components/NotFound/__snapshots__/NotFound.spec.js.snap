// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFound
    prop="NotFound"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span>
        NotFound
      </span>,
      "className": "NotFound",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "NotFound",
      },
      "ref": null,
      "rendered": "NotFound",
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span>
          NotFound
        </span>,
        "className": "NotFound",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "NotFound",
        },
        "ref": null,
        "rendered": "NotFound",
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
